### PR TITLE
fstack: Refactor setup_task_{filter,handle}

### PR DESCRIPTION
**Pull request by Taeung**
Extracting duplicated code of setup_task_filter()
and setup_task_handle(), create new prepare_task_handle()

Signed-off-by: Taeung Song <treeze.taeung@gmail.com>

### CODE REVIEW

**Comment by Taeung** [comment]
@namhyung , In addtion, I thought it is better to make `setup_task_filter()` and
`setup_task_handle()` static functions like below,
because currently the two functions are used in only utils/fstack.c, no ?
```patch
diff --git a/utils/fstack.c b/utils/fstack.c
index 0746381..4b3fb4b 100644
--- a/utils/fstack.c
+++ b/utils/fstack.c
@@ -37,8 +37,8 @@ struct ftrace_task_handle *get_task_handle(struct ftrace_file_handle *handle,
         return NULL;
 }
 
-void setup_task_handle(struct ftrace_file_handle *handle,
-                       struct ftrace_task_handle *task, int tid)
+static void setup_task_handle(struct ftrace_file_handle *handle,
+                              struct ftrace_task_handle *task, int tid)
 {
         int i;
         char *filename;
@@ -140,7 +140,7 @@ static void update_first_timestamp(struct ftrace_file_handle *handle,
  * This function sets up task filters using @tid_filter.
  * Tasks not listed will be ignored.
  */
-void setup_task_filter(char *tid_filter, struct ftrace_file_handle *handle)
+static void setup_task_filter(char *tid_filter, struct ftrace_file_handle *handle)
 {
         int i, k;
         int nr_filters = 0;
diff --git a/utils/fstack.h b/utils/fstack.h
index 77be98a..731efd8 100644
--- a/utils/fstack.h
+++ b/utils/fstack.h
@@ -96,8 +96,6 @@ enum argspec_string_bits {
 extern bool fstack_enabled;
 extern bool live_disabled;
 
-void setup_task_handle(struct ftrace_file_handle *handle,
-                       struct ftrace_task_handle *task, int tid);
 struct ftrace_task_handle *get_task_handle(struct ftrace_file_handle *handle,
                                            int tid);
 void reset_task_handle(struct ftrace_file_handle *handle);
@@ -127,7 +125,6 @@ static inline bool is_kernel_record(struct ftrace_task_handle *task,
         return rec == &task->kstack;
 }
 
-void setup_task_filter(char *tid_filter, struct ftrace_file_handle *handle);
 void setup_fstack_args(char *argspec, char *retspec,
                        struct ftrace_file_handle *handle, bool auto_args);
 int fstack_setup_filters(struct opts *opts, struct ftrace_file_handle *handle);

```

File: utils/fstack.c [hunk 1]
```diff
@@ -77,8 +58,6 @@ void setup_task_handle(struct ftrace_file_handle *handle,
 	max_stack = handle->hdr.max_stack;
 	task->func_stack = xcalloc(1, sizeof(*task->func_stack) * max_stack);
 
-	setup_rstack_list(&task->rstack_list);
-
```
Comment (namhyung) [inline utils/fstack.c]: Where did you call `prepare_task_handle()`?

File: utils/fstack.c [hunk 2]
```diff
@@ -174,6 +177,8 @@ void setup_task_filter(char *tid_filter, struct ftrace_file_handle *handle)
 		int tid = handle->info.tids[i];
 		struct ftrace_task_handle *task = &handle->tasks[i];
 
+		prepare_task_handle(handle, task, tid);
+
```
Comment (Taeung) [inline utils/fstack.c]: Here, I call `prepare_task_handle()`

Comment (namhyung) [reply utils/fstack.c]: Ah, I was confused due to the recent code change.  The setup_task_handle() is not called from outside anymore.  So yes, as you said, we should make it static.  Sorry for the noise.

**Comment by Taeung** [comment]
@namhyung , I tested this PR with test script related to 'tid'
```
$ ./runtest.py tid
Test case                 pg             finstrument-fu
------------------------: O0 O1 O2 O3 Os O0 O1 O2 O3 Os
091 replay_tid          : OK OK OK OK OK OK OK OK OK OK
092 report_tid          : OK OK OK OK OK OK OK OK OK OK
095 graph_tid           : OK OK OK OK OK OK OK OK OK OK
098 dump_tid            : OK OK OK OK OK OK OK OK OK OK
111 kernel_tid          : SK SK SK SK SK SK SK SK SK SK
129 session_tid         : OK OK OK OK OK OK OK OK OK OK
137 kernel_tid_update   : SK SK SK SK SK SK SK SK SK SK
```

File: utils/fstack.c [hunk 3]
```diff
@@ -202,11 +198,9 @@ void setup_task_filter(char *tid_filter, struct ftrace_file_handle *handle)
 				fclose(task->fp);
 				task->fp = NULL;
 			}
-			free(filename);
 			continue;
 		}
 
-		task->tid = tid;
 		setup_task_handle(handle, task, tid);
```
Comment (Taeung) [inline utils/fstack.c]: > The setup_task_handle() is not called from outside anymore.

Here, setup_task_handle() is called, @namhyung

**Comment by namhyung** [comment]
Could you please add a commit to make them static?

**Comment by Taeung** [comment]
Sure, I did it, please check this change 😄

**Pull request merged**